homeassistant/components/tradfri/__init__.py

```diff
@@ -103,8 +106,10 @@ async def async_setup(hass: HomeAssistant, config: ConfigType):
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Create a gateway."""
     # host, identity, key, allow_tradfri_groups
-    tradfri_data = hass.data.setdefault(DOMAIN, {})[entry.entry_id] = {}
-    listeners = tradfri_data[LISTENERS] = []
+    tradfri_data: dict[str, list[Any] | None] = {}
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = tradfri_data
+    listeners = []
+    tradfri_data[LISTENERS] = []
```

Comment:
`listeners` need to be the same list as the one stored in hass.data.

Reply:
Ups, forgot to merge those 2 lines after experimenting. Done.